test/test-eth-trezor-keyring.js

```diff
@@ -128,16 +132,22 @@ describe('TrezorKeyring', function () {
     })
 
     it('should call TrezorConnect.getPublicKey if we dont have a public key', async function () {
+      const getPublicKeyStub = sinon.stub(TrezorConnect, 'getPublicKey').callsFake(
+        () => Promise.resolve({}),
+      )
       chai.spy.on(TrezorConnect, 'getPublicKey')
+
       keyring.hdk = new HDKey()
       try {
         await keyring.unlock()
       } catch (e) {
-        // because we're trying to open the trezor popup in node
-        // it will throw an exception
-      } finally {
-        expect(TrezorConnect.getPublicKey).to.have.been.called()
+        // Since we only care about ensuring our function gets called,
+        // we want to ignore warnings due to stub data
       }
+
+      expect(TrezorConnect.getPublicKey).to.have.been.called()
+
+      getPublicKeyStub.restore()
```

Comment:
Instead of restoring here, I'd recommend calling `sinon.restore()` in an `afterEach` hook. That would ensure it runs even if the test crashes.

See: https://sinonjs.org/releases/v9.2.3/general-setup/

Reply:
Now that `sinon.restore()` is in the `afterEach` hook, this call is redundant. `sinon.restore()` will restore _all stubs_, so all `.restore()` calls on individual stubs can now be removed.

Reply:
Yep, my bad, updated.